c2_components/src/mfx_c2_encoder_component.cpp

```diff
@@ -1779,7 +1824,7 @@ c2_status_t MfxC2EncoderComponent::Queue(std::list<std::unique_ptr<C2Work>>* con
                 DoWork(std::move(work));
             } );
 
-            if(eos || !m_pendingWorks.empty()) {
+            if(eos /*|| !m_pendingWorks.empty()*/) {
```

Comment:
I think if input_buffers has data coming in, the pendingWorks shouldn't do drain().
It may cause errors.

Reply:
This is a potential issue, we can change it for now, but please recall if any related regression found.